File: the-xray-tracer/typescript/lib/the-dynamo-flow-stack.ts
import * as cdk from '@aws-cdk/core';
import lambda = require('@aws-cdk/aws-lambda');
import dynamodb = require('@aws-cdk/aws-dynamodb');
import sns_sub = require('@aws-cdk/aws-sns-subscriptions');
import sns = require('@aws-cdk/aws-sns');

export interface DynamoFlowStackProps extends cdk.StackProps{
    readonly snsTopicARN: string;
}

export class TheDynamoFlowStack extends cdk.Stack {
    dynamoLambda: lambda.Function;
    constructor(scope: cdk.Construct, id: string, props: DynamoFlowStackProps) {
        super(scope, id, props);

        //DynamoDB Table
        const table = new dynamodb.Table(this, 'Hits', {
            partitionKey: { name: 'path', type: dynamodb.AttributeType.STRING }
        });

        // defines an AWS Lambda resource
        this.dynamoLambda = new lambda.Function(this, 'DynamoLambdaHandler', {
            runtime: lambda.Runtime.NODEJS_12_X,
            code: lambda.Code.fromAsset('lambda-fns'),
            handler: 'dynamo.handler',
            environment: {
                HITS_TABLE_NAME: table.tableName
            },
            tracing: lambda.Tracing.ACTIVE
        });

        // grant the lambda role read/write permissions to our table
        table.grantReadWriteData(this.dynamoLambda);

        let topic = sns.Topic.fromTopicArn(this, 'SNSTopic', props.snsTopicARN);
        topic.addSubscription(new sns_sub.LambdaSubscription(this.dynamoLambda));
    }
}

File: the-xray-tracer/python/the_xray_tracer/the_dynamo_flow_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_sns as sns,
    aws_sns_subscriptions as subscriptions,
    aws_dynamodb as dynamo_db,
    core
)


class TheDynamoFlowStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, sns_topic_arn: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # DynamoDB Table
        table = dynamo_db.Table(self, "Hits",
                                partition_key=dynamo_db.Attribute(name="path", type=dynamo_db.AttributeType.STRING)
                                )

        dynamo_lambda = _lambda.Function(self, "DynamoLambdaHandler",
                                         runtime=_lambda.Runtime.NODEJS_12_X,
                                         handler="dynamo.handler",
                                         code=_lambda.Code.from_asset("lambda_fns"),
                                         tracing=_lambda.Tracing.ACTIVE,
                                         environment={
                                             "HITS_TABLE_NAME": table.table_name
                                         }
                                         )
        # grant the lambda role read/write permissions to our table
        table.grant_read_write_data(dynamo_lambda)

        topic = sns.Topic.from_topic_arn(self, 'SNSTopic', sns_topic_arn)
        topic.add_subscription(subscriptions.LambdaSubscription(dynamo_lambda))
